### PR TITLE
feat(gcp): add compute_instance_group_manager

### PR DESCRIPTION
**Pull request by kevinkengne**
**Objective:**

Add support for `compute_instance_group_manager` and `compute_region_instance_group_manager`.

**Status:**

- [x] Added to registry.go
- [x] Added resource file
- [x] Added integration tests
- [x] Added unit tests if applicable - Not applicable

**Issues:**

Currently as suggested by @tim775, only one apllication `version` is supported per instance group. Support for multiple version per instance group could be added in a separate PR.

**Useful links:**

Pricing: 

https://cloud.google.com/compute/vm-instance-pricing
https://cloud.google.com/compute/disks-image-pricing#disk
https://cloud.google.com/compute/gpus-pricing

Terraform: 

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_group_manager
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_instance_group_manager
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance_template

### CODE REVIEW

File: internal/providers/terraform/google/compute_instance_group_manager.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetComputeInstanceGroupManagerRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:                "google_compute_instance_group_manager",
+		RFunc:               NewComputeInstanceGroupManager,
+		ReferenceAttributes: []string{"version.0.instance_template"},
```
Comment (tim775) [inline internal/providers/terraform/google/compute_instance_group_manager.go:12]:
```suggestion
		Name:                "google_compute_instance_group_manager",
		RFunc:               NewComputeInstanceGroupManager,
		Notes:               []string{"Multiple versions are not supported."},
		ReferenceAttributes: []string{"version.0.instance_template"},
```

File: internal/providers/terraform/google/compute_region_instance_group_manager.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetComputeRegionInstanceGroupManagerRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:                "google_compute_region_instance_group_manager",
+		RFunc:               NewComputeRegionInstanceGroupManager,
+		ReferenceAttributes: []string{"version.0.instance_template"},
```
Comment (tim775) [inline internal/providers/terraform/google/compute_region_instance_group_manager.go:12]:
```suggestion
		Name:                "google_compute_region_instance_group_manager",
		RFunc:               NewComputeRegionInstanceGroupManager,
		Notes:               []string{"Multiple versions are not supported."},
		ReferenceAttributes: []string{"version.0.instance_template"},
```

**Review by tim775 — CHANGES_REQUESTED** [review]
This is great Kevin, thanks!  Looks like it turned into a bit of a beast.  I added a couple quick suggestions to note that we don't support multiple versions.

**Comment by kevinkengne** [comment]
> This is great Kevin, thanks! Looks like it turned into a bit of a beast. I added a couple quick suggestions to note that we don't support multiple versions.

Thanks Tim! I've added the changes

**Review by tim775 — APPROVED** [review]

**Pull request merged**